src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -329,22 +335,52 @@ await CallWithRequestTracing(
                                 }
                             }
 
-                            if (refreshAll)
+                            if (!refreshAll)
```

Comment:
I feel handling feature flag refresh seperatedly from handling key value refresh will make the code flow more clear.

Like this
```
if (_options.RefreshOptions.RegisterAllEnabled) {
	// Check whether selected key value collection changed (using page etag)
	// if yes, load selected key value
}
else {
	// Check whether registered watchers （sentinel keys) changed
	for each changes:
		update the value of the watched key

	if any watched key can refresh all:
		load selected key value
}


// Check whether selected feature flag collection changes (using page etag)
// if yes, load feature flags
```

Reply:
Only thing is I think refreshall kind of should be first since if it refreshes everything then we shouldn't check the rest, but I can still try to rearrange to make things clear